test/blackbox-tests/test-cases/enabled_if/eif-exe-name-collision-same-folder.t

```diff
@@ -0,0 +1,59 @@
+Using same executable name in two contexts, where the executables are defined
+in the same dune file
+
+  $ cat > dune-project << EOF
+  > (lang dune 3.13)
+  > (package (name bar) (allow_empty))
+  > (package (name baz) (allow_empty))
+  > EOF
+
+  $ cat > dune-workspace << EOF
+  > (lang dune 3.13)
+  > 
+  > (context default)
+  > 
+  > (context
+  >  (default
+  >   (name alt-context)))
+  > EOF
+  $ cat > dune << EOF
+  > (executable
+  >  (name foo)
+  >  (enabled_if (= %{context_name} "default")))
+  > (executable
+  >  (name foo)
+  >  (enabled_if (= %{context_name} "alt-context")))
+  > EOF
+  $ cat > foo.ml <<EOF
+  > let () = print_endline "foo"
+  > EOF
+
+  $ dune build
+  File "dune", line 4, characters 0-72:
+  4 | (executable
+  5 |  (name foo)
+  6 |  (enabled_if (= %{context_name} "alt-context")))
+  Error: Executable "foo" appears for the second time in this directory
+  [1]
+
+For public exes
+
+  $ cat > dune << EOF
+  > (executable
+  >  (public_name foo)
```

Comment:
No need to test public exes separately. It's the exact same thing.

Reply:
I copied over from library tests 😅  cleaned up in https://github.com/ocaml/dune/pull/10211/commits/0f9bbadda48585eb2e4c9aa9186ba154e0f82cde.